network/Message.cpp

```diff
@@ -179,12 +181,15 @@ Message GameStartMessage(bool single_player_game, int empire_id,
                          CombatLogManager& combat_logs, const SupplyManager& supply,
                          const std::map<int, PlayerInfo>& players,
                          GalaxySetupData galaxy_setup_data,
-                         bool use_binary_serialization)
+                         bool use_binary_serialization, int zlib_level)
 {
     std::ostringstream os;
     {
+        boost::iostreams::filtering_ostream zos;
```

Comment:
In my tests on rather large archives using `filtering_ostream`, it was sometimes necessary to add `if (!zos.strict_sync()) zos.reset();` or similar before accessing the streamed data eg. via `.str()`

Reply:
I've added it to code.